ludwig/utils/upload_utils.py

```diff
@@ -205,3 +249,144 @@ def upload(
+                implementations. Defaults to None.
+
+        Raises:
+            AssertionError: If the repo_id has non-url safe characters.
+        """
+        assert len(repo_id) <= 255, "`repo_id` must be 255 characters or less."
+        BaseModelUpload._validate_upload_parameters(
+            repo_id,
+            model_path,
+            repo_type,
+            private,
+            commit_message,
+            commit_description,
+        )
+
+    def upload(
+        self,
+        repo_id: str,
+        model_path: str,
+        commit_description: Optional[str] = None,
+        dataset_file: Optional[str] = None,
+        dataset_name: Optional[str] = None,
+        **kwargs,
+    ) -> bool:
+        """Create an empty repo in Predibase and upload trained model artifacts to that repo.
+
+        Args:
+            model_path (`str`):
+                The path of the saved model. This is the top level directory where
+                the models weights as well as other associated training artifacts
+                are saved.
+            repo_name (`str`):
+                A repo name.
+            repo_description (`str` *optional*):
+                The description of the repo.
+            dataset_file (`str` *optional*):
+                The path to the dataset file. Required if `service` is set to
+                `"predibase"` for new model repos.
+            dataset_name (`str` *optional*):
+                The name of the dataset. Used by the `service`
+                `"predibase"`. Falls back to the filename.
+        """
+        # Validate upload parameters are in the right format
+        Predibase._validate_upload_parameters(
+            repo_id,
+            model_path,
+            None,
+            False,
+            "",
+            commit_description,
+        )
+
+        # Upload the dataset to Predibase
+        try:
+            dataset = self.pc.upload_dataset(file_path=dataset_file, name=dataset_name)
+        except Exception as e:
+            raise RuntimeError("Failed to upload dataset to Predibase") from e
+            return True
+
+        # Create empty model repo using repo_name, but it is okay if it already exists.
+        try:
+            repo = self.pc.create_model_repo(
+                name=repo_id,
+                description=commit_description,
+                exists_ok=True,
+            )
+        except Exception as e:
+            raise RuntimeError("Failed to create repo in Predibase") from e
+            return True
+
+        # Upload the zip file to Predibase
+        try:
+            self.pc.upload_model(
+                repo=repo,
+                model_path=model_path,
+                dataset=dataset,
+            )
+        except Exception as e:
+            raise RuntimeError("Failed to upload model to Predibase") from e
+            return True
+
+        logger.info(f"Model uploaded to Predibase with repository name `{repo_id}`")
+        return False
```

Comment:
QQ: why does this function return False on success and True on error? Can you add a comment about the return type to the docstring? My intuition here would be that it should return True on success, for what it's worth.

Reply:
Agreed, I am also confused by this

Reply:
Good callout! I mixed up my `True` and `False` because Predibase works a little different than HF. Fixing now!

Reply:
In addition, I've removed the other return statements as they're unreachable.